### PR TITLE
4480: Shrine demographics diagnose  and  4850: shrine flip flop 

### PR DESCRIPTION
**Pull request by lav-patel**

### CODE REVIEW

File: Oracle/shrine_AdapterMappings.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+set -x
+set -e
+SID=$1
+USERNAME=$2
+PASSWORD=$3
```
Comment (dckc) [inline Oracle/shrine_AdapterMappings.sh:6]: passwords on the command line are visible to any user on the host. In many (most?) places we use environment variables, which are only visible to the user that owns the process.

I don't suppose we have made this a policy yet, though; I don't see documentation of it our wiki.

I wonder what Matt and/or Daryl think.

File: Oracle/shrine_AdapterMappings.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+########################################################################
+# Coppying NACCR onotologoies and Generating SHRRINE ONT Mapping 
+########################################################################
+
+sqlplus -S $USERNAME/$PASSWORD@$SID @shrine_AdapterMappings.sql
+
+##########################################################################
+# exporting shrine ont mapping as CSV and then comnine with manual mapping
```
Comment (dckc) [inline Oracle/shrine_AdapterMappings.sh:14]: typo: combine

File: Oracle/shrine_export_AdapterMapping.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+set linesize 32000
+set pagesize 0   -- No header rows
+set trimspool on -- remove trailing blanks
+set feedback off
+-- select '"'|| SCIHLS_PATH || '","'|| HERON_PATH || '"' from  SHRINE_ONT.TEMP_SHRINE_MAPPING order by SCIHLS_PATH;
+-- some row has quatation in theri values, That is why I need to run sql statment.
+spool AdapterMappings.csv
+select '"'|| SCIHLS_PATH || '","'|| HERON_PATH || '"' from  SHRINE_ONT.TEMP_SHRINE_MAPPING
```
Comment (dckc) [inline Oracle/shrine_export_AdapterMapping.sql:8]: Are we guaranteed that there are no double-quotes in SCILHS_PATH?

Using sqlplus to write CSV files is pretty risky. I hope we can move toward using [Ant SQL Task](https://ant.apache.org/manual/Tasks/sql.html). But I guess if this is working, I can live with it.

File: Oracle/shrine_AdapterMappings.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+set -x
+set -e
+SID=$1
+USERNAME=$2
+PASSWORD=$3
+########################################################################
+# Coppying NACCR onotologoies and Generating SHRRINE ONT Mapping
```
Comment (dckc) [inline Oracle/shrine_AdapterMappings.sh:8]: typo: SHRINE

File: Oracle/shrine_AdapterMappings.sql [hunk 5]
```diff
@@ -0,0 +1,60 @@
+-- Drop old shrine NAACCR_ONTOLOGY and built fresh shrine NAACCR_ONTOLOGY
+-- But Not going to build new shrine (shrin_ont.shrine) ontolgies everytime,
```
Comment (dckc) [inline Oracle/shrine_AdapterMappings.sql:2]: typo: shrine_ont

Comment (dckc) [reply Oracle/shrine_AdapterMappings.sql:2]: typo: ontologies

**Review by dckc — APPROVED** [review]
None of my comments is critical, but I'd like Daryl's input.

File: Oracle/shrine_AdapterMappings.sh (hunk 1 above)
Comment (archaean) [reply Oracle/shrine_AdapterMappings.sh:6]: Worth 15 minutes of our time to see if taking in the password environment variable works as opposed to the password itself.

Comment (archaean) [reply Oracle/shrine_AdapterMappings.sh:6]: On the other hand it goes on the command line for sqlplus on line:17

**Pull request merged**